unblob/pool.py

```diff
@@ -1,11 +1,30 @@
 import abc
 import multiprocessing as mp
+import os
+import signal
 import sys
 from multiprocessing.queues import JoinableQueue
 from typing import Any, Callable, Union
 
+from structlog import get_logger
+
 from .logging import multiprocessing_breakpoint
 
+logger = get_logger()
+
+
+class ShutDownRequired(Exception):
+    pass
+
+
+def _handle_signal(signum: int, _frame):
```

Comment:
we should also call the original signal handlers from ours

Reply:
I just re-set the signal handlers at the end of the `pool.close` to point back to the original handlers.

But you are right, we need to re-raise (re-fire?) those signals as well to let the original handlers handle it as they please. I made a separate commit about this to show its importance.